practice/validator/site.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+/**
+ * Tests whether the string matches the regular expression
+ * @param  {string} testString The string to test
+ * @param  {reg exp} regExp    The regular expression.  Not a string.
+ * @return {boolean}           True if there is a match, false otherwise.
+ */
+function stringValidator(testString, regExp) {
+  return regExp.test(testString);
+}
+
+/**
+ * The the string entered on the form.
+ * @param  {jQuery selector} entryItem The selector for a text input.
+ * @return {string}                    The string entered on the form.
+ */
+function getFieldEntry(entryItem) {
+  return entryItem.val();
+}
+
+/**
+ * displayInvalid() highlights the input box and un-hides the warning message.
+ */
+function displayInvalid(entryItem) {
+  entryItem.addClass('invalid');
+  $(entryItem).next().addClass('visible-warning');
+}
+
+/**
+* displayValid() un-highlights the input box and hides the warning message.
+ */
+function displayValid(entryItem) {
+  entryItem.removeClass();
+  $(entryItem).next().removeClass();
+}
+/**
+ * Check whether the input string is a match and changes class to invalid if not
+ * @param  {jQuery selector} entryItem The selector on the form for an input
+ * @param  {reg exp} regExp            Regular expression to test the match
+ */
+function runValidator(entryItem, regExp) {
+  var entryString = getFieldEntry(entryItem);
+  if(stringValidator(entryString, regExp)) {
+    displayValid(entryItem);
+  } else {
+    displayInvalid(entryItem);
+  }
+}
+
+/**
+ * Main Name validator, for the name entry on the form.
+ */
+function runNameValidator() {
+  var nameEntryItem = $('#name-input');
+  var nameRegExp = /^[a-z]*(\s[a-z]*)?$/i;
+  runValidator(nameEntryItem, nameRegExp);
+}
+
+/**
+ * Main Date of Birth validator, for the DoB entry on the form.
+ */
+function runDobValidator() {
+  var dobEntryItem = $('#dob-input');
+  var dobRegExp = /^\d{0,4}$|^\d{4}-\d{0,2}$|^\d{4}-\d{2}-\d{0,2}$/;
+  runValidator(dobEntryItem, dobRegExp);
+}
+
+/**
+ * Main Phone number validator, for the Phone Num entry on the form.
+ */
+function runPhoneValidator() {
+  var phoneEntryItem = $('#phone-input');
+  var phoneRegExp = /^\d{0,3}$|^\d{3}-\d{0,3}$|^\d{3}-\d{3}-\d{0,4}$/;
+  runValidator(phoneEntryItem, phoneRegExp);
+}
+
+/**
+ * runFinalNameChecker() checks to see if the entered name is valid.
+ * Returns boolean, true if name is valid.
+ */
+function runFinalNameChecker() {
+  var nameEntryItem = $('#name-input');
+  var nameRegExp = /^[a-z]+\s[a-z]+$/i;
+  var nameString = getFieldEntry(nameEntryItem);
+  return nameRegExp.test(nameString);
+}
+
```

Comment:
This function is very similar to run name Validator, (and all the other Validators and Final Checkers), particularly in defining the variables. I know the variables have to be defined at the instance of the function (because the event is continuous), but I bet you could reduce the pattens even further with a variable defining function for xItem, xExp, and xString given x, (3 variable defining functions for each Item, Exp, and String).